src/legacy/core_plugins/kibana/migrations.js

```diff
@@ -145,6 +145,38 @@ export const migrations = {
       } catch (e) {
         throw new Error(`Failure attempting to migrate saved object '${doc.attributes.title}' - ${e}`);
       }
+    },
+    '7.1.0': (doc) => {
+      // [TSVB] Migrate percentile-rank aggregation (value -> values)
+      const visStateJSON = get(doc, 'attributes.visState');
+
+      if (visStateJSON) {
+        let visState;
+        try {
+          visState = JSON.parse(visStateJSON);
+        } catch (e) {
+          // Let it go, the data is invalid and we'll leave it as is
+        }
+        if (visState && visState.type === 'metrics') {
+          const series = get(visState, 'params.series') || [];
+
+          series.forEach(series => {
+            if (!isEmpty(series.metrics)) {
+              series.metrics.forEach(metric => {
+                if (metric.type === 'percentile_rank' && metric.value) {
```

Comment:
I'm not entirely sure what metric.value's type is, but if it's numeric, this will incorrectly skip documents with a metric.value of 0, for example. I *think* this should be: `if (metric.type === 'percentile_rank' && metric.hasOwnProperty('value')`